lib/exprengine.cpp

```diff
@@ -588,7 +588,7 @@ namespace {
         std::string str() const {
             std::ostringstream ret;
             std::map<std::string, ExprEngine::ValuePtr> vars;
-            for (const auto mem: memory) {
+            for (const auto &mem: memory) {
```

Comment:
it's only 12 bytes to copy so that is not much. but without the reference, the shared_ptr will cause a reference counting.

I made a test for this particular case and here it was faster to pass by reference as far as I see.

Reply:
Hm, interesting. I don't think about shared pointers. But that's right, copying of `shared_ptr` leads to reference counting.
I do the simple test too: https://gist.github.com/jubnzv/b890f024288bcd084f406ac5368d7220
In my machine the case with passing by reference is faster about 0.75 times.